pkg/build/strategies/sti/postexecutorstep.go

```diff
@@ -251,7 +251,7 @@ func (step *startRuntimeImageAndUploadFilesStep) execute(ctx *postExecutorStepCo
 		return fmt.Errorf("Couldn't get working dir of %q image: %v", image, err)
 	}
 
-	commandBaseDir := workDir
+	commandBaseDir := filepath.Join(workDir, "scripts")
 	useExternalAssembleScript := step.builder.externalScripts[api.AssembleRuntime]
 	if !useExternalAssembleScript {
```

Comment:
i don't understand why this if-check is not being triggered for the scenario you described and setting commandBaseDir correctly so we run the script that is already part of the image?